### PR TITLE
Add cbor decoding for DealProposal and MarketDealNotifyParams

### PR DESCRIPTION
**Pull request by shrenujbansal**
:link: [zboto Link](https://app.clickup.com/t/8677eddp2)

### CODE REVIEW

**Comment by shrenujbansal** [comment]
> @shrenujbansal Do you think it is possible to add tests for this ?

Done

File: contracts/v0.8/cbor/MarketCbor.sol [hunk 1]
```diff
@@ -280,4 +282,69 @@ library MarketCBOR {
...
+    function deserializeMarketDealNotifyParams(bytes memory rawResp) internal pure returns (MarketTypes.MarketDealNotifyParams memory ret) {
+        uint byteIdx = 0;
+        uint len;
+
+        (len, byteIdx) = rawResp.readFixedArray(byteIdx);
+        assert(len == 2);
+
+        (ret.dealProposal, byteIdx) = rawResp.readBytes(byteIdx);
+        (ret.dealId, byteIdx) = rawResp.readUInt64(byteIdx);
+
```
Comment (raulk) [inline contracts/v0.8/cbor/MarketCbor.sol:295]: extra line

File: contracts/v0.8/cbor/MarketCbor.sol [hunk 2]
```diff
@@ -280,4 +282,69 @@ library MarketCBOR {
...
+        ret.storage_price_per_epoch = storage_price_per_epoch_bytes.deserializeBigInt();
+
+        bytes memory provider_collateral_bytes;
+        (provider_collateral_bytes, byteIdx) = rawResp.readBytes(byteIdx);
+        ret.provider_collateral = provider_collateral_bytes.deserializeBigInt();
+
+        bytes memory client_collateral_bytes;
+        (client_collateral_bytes, byteIdx) = rawResp.readBytes(byteIdx);
+        ret.client_collateral = client_collateral_bytes.deserializeBigInt();
+
```
Comment (raulk) [inline contracts/v0.8/cbor/MarketCbor.sol:348]: extra line

File: contracts/v0.8/tests/marketcbor.test.sol [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    using MarketCBOR for *;
+
+    function bigIntEqual(BigInt memory n1, BigInt memory n2) internal pure {
+        require(keccak256(n1.val)==keccak256(n2.val));
+        require(n1.neg == n2.neg);
+    }
+
+    function testDealProposalSerDes() view public {
+
```
Comment (raulk) [inline contracts/v0.8/tests/marketcbor.test.sol:18]: extra line

File: contracts/v0.8/tests/marketcbor.test.sol [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        require(proposal.verified_deal == deserProposal.verified_deal, "unequal verified_deal");
+        require(keccak256(proposal.client) == keccak256(deserProposal.client), "unequal client");
+        require(keccak256(proposal.provider) == keccak256(deserProposal.provider), "unequal provider");
+        require(keccak256(abi.encodePacked(proposal.label)) == keccak256(abi.encodePacked(deserProposal.label)), "unequal label");
+        require(proposal.start_epoch == deserProposal.start_epoch, "unequal start_epoch");
+        require(proposal.end_epoch == deserProposal.end_epoch, "unequal end_epoch");
+        bigIntEqual(proposal.storage_price_per_epoch, deserProposal.storage_price_per_epoch);
+        bigIntEqual(proposal.provider_collateral, deserProposal.provider_collateral);
+        bigIntEqual(proposal.client_collateral, deserProposal.client_collateral);
+
```
Comment (raulk) [inline contracts/v0.8/tests/marketcbor.test.sol:46]: extra line

File: contracts/v0.8/tests/marketcbor.test.sol [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import { MarketTypes } from "../types/MarketTypes.sol";
+import { MarketCBOR } from "../cbor/MarketCbor.sol";
+import { BigInt } from "../cbor/BigIntCbor.sol";
+
+contract MarketCBORTest {
+
+    using MarketCBOR for *;
+
+    function bigIntEqual(BigInt memory n1, BigInt memory n2) internal pure {
+        require(keccak256(n1.val)==keccak256(n2.val));
```
Comment (raulk) [inline contracts/v0.8/tests/marketcbor.test.sol:13]: code style: operator spacing

File: contracts/v0.8/tests/marketcbor.test.sol [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+import { MarketTypes } from "../types/MarketTypes.sol";
+import { MarketCBOR } from "../cbor/MarketCbor.sol";
+import { BigInt } from "../cbor/BigIntCbor.sol";
+
+contract MarketCBORTest {
+
+    using MarketCBOR for *;
+
+    function bigIntEqual(BigInt memory n1, BigInt memory n2) internal pure {
```
Comment (raulk) [inline contracts/v0.8/tests/marketcbor.test.sol:12]:
```suggestion
    function requireBigIntEqual(BigInt memory n1, BigInt memory n2) internal pure {
```

Or make it return a boolean and perform the require outside.

**Comment by raulk** [comment]
@emmanuelm41 is this failing because @shrenujbansal is an outside contributor, and therefore GitHub Actions doesn't use the preconfigured secrets?

**Comment by snissn** [comment]
This pr creates a struct MarketDealNotifyParams but it has a param with bytes:
```
struct MarketDealNotifyParams {
        bytes dealProposal;
        uint64 dealId;
    }
```
Can we do this with the convention we've used earlier?
```
struct MarketDealNotifyParams {
        DealProposal proposal;
        uint64 dealId;
    }
```

**Comment by emmanuelm41** [comment]
@raulk things are failing now because the branch is outdated. The deploy tests are failing because he is an outsider contributor, indeed. But those are now important in this case. Let's update the branch and merge it. We will create the rust test later.

**Comment by snissn** [comment]
I'm confused here - is this necessary to include in this library, or is it implementation specific and able to be defined in the client-contract codebase? @shrenujbansal If dealID is something that you created for book keeping in your client, I'm not sure this struct needs to be in this current library and can easily be added to client-contract like this: https://github.com/lotus-web3/client-contract/pull/11/commits/9436742a3b356a193f209b887a6939ca2a59663d

also i see conflicted references to dealId as a uint64 and as a bytes32 hash

**Comment by snissn** [comment]
thanks to @raulk and @shrenujbansal 's explanation this is good to be merged, thanks 👍

**Pull request closed without merging**